record-editor/src/app/record-editor/record-toolbar/record-toolbar.component.ts

```diff
@@ -133,19 +134,22 @@ export class RecordToolbarComponent extends SubscriberComponent
 
   private cleanupAndSaveRecord(record) {
     this.recordCleanupService.cleanup(record);
-    this.apiService
-      .saveRecord(record)
-      .subscribe(() => this.onSaveSuccess(), error => this.onSaveError(error));
+    this.apiService.saveRecord(record).subscribe(
+      () => this.onSaveSuccess(),
+      (error) => this.onSaveError(error)
+    );
   }
 
   private onSaveSuccess() {
     this.domUtilsService.unregisterBeforeUnloadPrompt();
-
-    if (this.pidType === 'conferences') {
+    this.toastrService.success('Changes are saved', 'Success');
+    if (this.pidType === 'conferences' || this.pidType === 'literature') {
       // direct assignment to window.location  doesn't compile with this version of TS
-      window.location.href = `/conferences/${this.record['control_number']}`;
-    } else {
-      this.toastrService.success('Changes are saved', 'Success');
+      setInterval(
```

Comment:
set inteval why? 🤔

Reply:
In order to show the success message. It will be confusing redirecting the page without displaying a feedback.

Reply:
I think it's ok, it only redirects on success, right, not in case of error? curators will learn that if they see the detailed view, it means that their changes have been applied.